libuuid/uuid.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Universally Unique identifiers library
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "include/uuid/uuid.h"
+
+
+static void uuid_useRand(uuid_t out)
+{
+	pid_t pid;
+	struct timespec ts;
+	int i, n;
+
+	/* on some targets clock interval is 10ms, the sleep is to prevent duplication */
+	usleep(10 * 1000);
+	clock_gettime(CLOCK_MONOTONIC, &ts);
+	pid = getpid();
+	/* only if (pid2 == 255 + n*pid1) and the first 24 bits of tv_sec were the same it could be the same seed */
+	srand((unsigned int)((ts.tv_sec & 0x0FFF) | (pid << 24)));
+
+	/* to prevent same numbers when using the same seed */
+	n = (int)(ts.tv_nsec / 100);
+	for (i = 0; i < n; i++) {
+		rand();
+	}
+
+	for (i = 0; i < 16; i++) {
+		out[i] = (unsigned char)rand();
+	}
+}
+
+
+void uuid_unparse(const uuid_t uu, char *out)
+{
+	int i;
+	char *temp;
+
+	if ((uu == NULL) || (out == NULL)) {
+		return;
+	}
```

Comment:
No need for defensiveness - this is a lib function. If some passes (void *)1 as an argument it will still crash. Applies to the whole PR.